pkg/antctl/raw/check/installation/command.go

```diff
@@ -0,0 +1,407 @@
+	Name        string
+	Kind        string
+	Image       string
+	Replicas    int
+	Port        int
+	Command     []string
+	Affinity    *corev1.Affinity
+	Tolerations []corev1.Toleration
+	Labels      map[string]string
+}
+
+func newDeployment(p deploymentParameters) *appsv1.Deployment {
+	if p.Replicas == 0 {
+		p.Replicas = 1
+	}
+	replicas32 := int32(p.Replicas)
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: p.Name,
+			Labels: map[string]string{
+				"name": p.Name,
+				"kind": p.Kind,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: p.Name,
+					Labels: map[string]string{
+						"name": p.Name,
+						"kind": p.Kind,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: p.Name,
+							Env: []corev1.EnvVar{
+								{Name: "PORT", Value: fmt.Sprintf("%d", p.Port)},
+							},
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: int32(p.Port)},
+							},
+							Image:           p.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         p.Command,
+						},
+					},
+					Affinity: p.Affinity,
+				},
+			},
+			Replicas: &replicas32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": p.Name,
+					"kind": p.Kind,
+				},
+			},
+		},
+	}
+}
+
+func NewTestContext(client k8sClientOperations, o *options) *testContext {
+	return &testContext{
+		client:          client,
+		antreaNamespace: o.antreaNamespace,
+		namespace:       generateRandomNamespace(postInstallationTestsNamespace),
+	}
+}
+
+func generateRandomNamespace(baseName string) string {
+	const letters = "abcdefghijklmnopqrstuvwxyz0123456789"
+	bytes := make([]byte, 5)
+	_, err := rand.Read(bytes)
+	if err != nil {
+		panic(err)
+	}
+	for i, b := range bytes {
+		bytes[i] = letters[b%byte(len(letters))]
+	}
+	return fmt.Sprintf("%s-%s", baseName, string(bytes))
+}
+
+const podReadyTimeout = 5 * time.Minute
+
+func (t *testContext) deleteDeployments(ctx context.Context, client k8sClientOperations) error {
+	t.Log("[%s] Deleting Post installation tests Deployments...", client.ClusterName())
+	client.GetClientSet().CoreV1().Namespaces().Delete(ctx, t.namespace, metav1.DeleteOptions{})
+	t.Log("[%s] Waiting for Namespace %s to disappear", client.ClusterName(), t.namespace)
+	err := wait.PollUntilContextTimeout(ctx, 1*time.Second, 5*time.Minute, true, func(ctx context.Context) (bool, error) {
+		_, err := client.GetClientSet().CoreV1().Namespaces().Get(ctx, t.namespace, metav1.GetOptions{})
+		if err != nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	return err
+}
+
+func (t *testContext) setup(ctx context.Context) error {
+	_, err := t.client.GetClientSet().AppsV1().DaemonSets(t.antreaNamespace).Get(ctx, agentDaemonSetName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Unable to determine status of Antrea DaemonSet: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("unable to determine status of Antrea DaemonSet: %w", err)
```
error string shouldn't be captilized, please revise all other cases.